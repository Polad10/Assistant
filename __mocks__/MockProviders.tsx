import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
