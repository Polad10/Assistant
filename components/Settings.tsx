import { useContext } from 'react'
import MainView from './MainView'
import { Icon, ListItem, Text } from '@rneui/themed'
import { ThemeContext } from '../contexts/ThemeContext'
import { useNavigation } from '@react-navigation/native'
import { RootStackScreenProps } from '../types/Navigation'
import { TouchableHighlight } from 'react-native-gesture-handler'
import { translate } from '../helpers/Translator'

export default function Settings() {
  const navigation = useNavigation<RootStackScreenProps<'Settings'>['navigation']>()
  const themeContext = useContext(ThemeContext)!

  return (
    <MainView style={{ paddingTop: 20 }}>
      <Text style={{ color: themeContext.neutral, fontSize: 16, opacity: 0.5, marginBottom: 5, marginLeft: 10 }}>
        {translate('general')}
      </Text>
      <ListItem
        containerStyle={{ backgroundColor: themeContext.secondary, borderColor: themeContext.border }}
        bottomDivider
        onPress={() => navigation.navigate('Languages')}
        Component={TouchableHighlight}
      >
        <Icon name='language-outline' type='ionicon' color={themeContext.neutral} />
        <ListItem.Content>
          <ListItem.Title style={{ color: themeContext.neutral }}>{translate('language')}</ListItem.Title>
        </ListItem.Content>
        <ListItem.Content right>
          <ListItem.Title style={{ color: themeContext.neutral, opacity: 0.5 }}>English</ListItem.Title>
        </ListItem.Content>
        <ListItem.Chevron size={20} />
      </ListItem>
      <ListItem
        containerStyle={{ backgroundColor: themeContext.secondary, borderColor: themeContext.border }}
        bottomDivider
        Component={TouchableHighlight}
      >
        <Icon name='color-palette-outline' type='ionicon' color={themeContext.neutral} />
        <ListItem.Content>
          <ListItem.Title style={{ color: themeContext.neutral }}>{translate('theme')}</ListItem.Title>
        </ListItem.Content>
        <ListItem.Content right>
          <ListItem.Title style={{ color: themeContext.neutral, opacity: 0.5 }}>Dark</ListItem.Title>
        </ListItem.Content>
        <ListItem.Chevron size={20} />
      </ListItem>
    </MainView>
  )
}
