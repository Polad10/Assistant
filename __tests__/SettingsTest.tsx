import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import Settings from '../components/Settings'

describe('Settings', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <Settings />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
