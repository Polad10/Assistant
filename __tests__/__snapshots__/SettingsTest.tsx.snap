// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#121212",
        "flex": 1,
      },
      {
        "paddingTop": 20,
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#faf8ff",
        "fontSize": 16,
        "marginBottom": 5,
        "marginLeft": 10,
        "opacity": 0.5,
      }
    }
  >
    General
  </Text>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={{}}
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#1a1a1a",
            "borderBottomWidth": 0.5,
            "borderColor": "#374151",
            "flexDirection": "row",
            "padding": 14,
          }
        }
        testID="RNE__LISTITEM__padView"
      >
        <View
          style={
            {
              "overflow": "hidden",
            }
          }
          testID="RNE__ICON__CONTAINER"
        >
          <View
            accessibilityRole="button"
            android_ripple={
              {
                "borderless": false,
                "color": "rgba(0, 0, 0, 0.3)",
                "radius": -5,
              }
            }
            disabled={false}
            testID="RNE__ICON__CONTAINER_ACTION"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              testID="RNE__ICON"
            >
              <Text />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          theme={
            {
              "colors": {
                "background": "#ffffff",
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": {
                  "android": {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ad1457",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
              "spacing": {
                "lg": 12,
                "md": 8,
                "sm": 4,
                "xl": 24,
                "xs": 2,
              },
            }
          }
        >
          <Text
            accessibilityRole="text"
            style={
              {
                "backgroundColor": "transparent",
                "color": "#faf8ff",
                "fontSize": 17,
              }
            }
            testID="listItemTitle"
          >
            Language
          </Text>
        </View>
        <View
          style={
            {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            [
              {
                "alignItems": "flex-end",
                "flex": 0.5,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          theme={
            {
              "colors": {
                "background": "#ffffff",
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": {
                  "android": {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ad1457",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
              "spacing": {
                "lg": 12,
                "md": 8,
                "sm": 4,
                "xl": 24,
                "xs": 2,
              },
            }
          }
        >
          <Text
            accessibilityRole="text"
            style={
              {
                "backgroundColor": "transparent",
                "color": "#faf8ff",
                "fontSize": 17,
                "opacity": 0.5,
              }
            }
            testID="listItemTitle"
          >
            English
          </Text>
        </View>
        <View
          style={
            {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            {
              "alignSelf": "center",
              "overflow": "hidden",
            }
          }
          testID="RNE__ICON__CONTAINER"
        >
          <View
            accessibilityRole="button"
            android_ripple={
              {
                "borderless": false,
                "color": "rgba(0, 0, 0, 0.3)",
                "radius": -5,
              }
            }
            disabled={false}
            testID="RNE__ICON__CONTAINER_ACTION"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              testID="RNE__ICON"
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    delayLongPress={600}
    enabled={true}
    exclusive={true}
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={{}}
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#1a1a1a",
            "borderBottomWidth": 0.5,
            "borderColor": "#374151",
            "flexDirection": "row",
            "padding": 14,
          }
        }
        testID="RNE__LISTITEM__padView"
      >
        <View
          style={
            {
              "overflow": "hidden",
            }
          }
          testID="RNE__ICON__CONTAINER"
        >
          <View
            accessibilityRole="button"
            android_ripple={
              {
                "borderless": false,
                "color": "rgba(0, 0, 0, 0.3)",
                "radius": -5,
              }
            }
            disabled={false}
            testID="RNE__ICON__CONTAINER_ACTION"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              testID="RNE__ICON"
            >
              <Text />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          theme={
            {
              "colors": {
                "background": "#ffffff",
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": {
                  "android": {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ad1457",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
              "spacing": {
                "lg": 12,
                "md": 8,
                "sm": 4,
                "xl": 24,
                "xs": 2,
              },
            }
          }
        >
          <Text
            accessibilityRole="text"
            style={
              {
                "backgroundColor": "transparent",
                "color": "#faf8ff",
                "fontSize": 17,
              }
            }
            testID="listItemTitle"
          >
            Theme
          </Text>
        </View>
        <View
          style={
            {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            [
              {
                "alignItems": "flex-end",
                "flex": 0.5,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          theme={
            {
              "colors": {
                "background": "#ffffff",
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": {
                  "android": {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ad1457",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
              "spacing": {
                "lg": 12,
                "md": 8,
                "sm": 4,
                "xl": 24,
                "xs": 2,
              },
            }
          }
        >
          <Text
            accessibilityRole="text"
            style={
              {
                "backgroundColor": "transparent",
                "color": "#faf8ff",
                "fontSize": 17,
                "opacity": 0.5,
              }
            }
            testID="listItemTitle"
          >
            Dark
          </Text>
        </View>
        <View
          style={
            {
              "paddingLeft": 16,
            }
          }
        />
        <View
          style={
            {
              "alignSelf": "center",
              "overflow": "hidden",
            }
          }
          testID="RNE__ICON__CONTAINER"
        >
          <View
            accessibilityRole="button"
            android_ripple={
              {
                "borderless": false,
                "color": "rgba(0, 0, 0, 0.3)",
                "radius": -5,
              }
            }
            disabled={false}
            testID="RNE__ICON__CONTAINER_ACTION"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              testID="RNE__ICON"
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;
